test/unit/uk/gov/hmrc/organisationsmatchingapi/services/CrnMatchingServiceSpec.scala

```diff
@@ -0,0 +1,150 @@
+  }
+
+  "crnMatchingService" should {
+    "basicMatch" should {
+      "match a basic address" in new Fixture {
+
+        val knownFactsData = models.CrnMatchData("mycrn", "myname", models.Address("foo", "bar", None, None, "code"))
+        val ifData = models.CrnMatchData("mycrn", "myname", models.Address("foo", "bar", None, None, "code"))
+
+        crnMatchingService.basicMatch(knownFactsData, ifData)
+      }
+
+      "fail to match where white space differs" in new Fixture {
+        val knownFactsData = models.CrnMatchData("mycrn ", " myname", models.Address("foo ", "b ar", None, None, "code"))
+        val ifData = models.CrnMatchData("mycrn", "myname", models.Address("foo", "bar", None, None, "code"))
+
+        crnMatchingService.basicMatch(knownFactsData, ifData) shouldBe false
+      }
+
+      "fail to match where punctuation differs" in new Fixture {
+        val knownFactsData = models.CrnMatchData("myc!rn ", " my'name", models.Address("foo .", "b ar", None, None, "code"))
+        val ifData = models.CrnMatchData("mycrn", "myname", models.Address("foo", "bar", None, None, "code"))
+
+        crnMatchingService.basicMatch(knownFactsData, ifData) shouldBe false
+      }
+
+      "fail to match where PO box differs" in new Fixture {
+        val knownFactsData = models.CrnMatchData("myc!rn ", " my'name", models.Address("P.O. Box787", "bar", None, None, "code"))
+        val ifData = models.CrnMatchData("mycrn", "myname", models.Address("PO Box787", "bar", None, None, "code"))
+
+        crnMatchingService.basicMatch(knownFactsData, ifData) shouldBe false
+      }
+    }
+
+    "cleanMatch" should {
+      "match a basic address" in new Fixture {
+        val knownFactsData = models.CrnMatchData("mycrn", "myname", models.Address("foo", "bar", None, None, "code"))
+        val ifData = models.CrnMatchData("mycrn", "myname", models.Address("foo", "bar", None, None, "code"))
+
+        crnMatchingService.cleanMatch(knownFactsData, ifData) shouldBe true
+      }
+
+      "match where white space differs" in new Fixture {
+        val knownFactsData = models.CrnMatchData("mycrn ", " myname", models.Address("foo ", "b ar", None, None, "code"))
+        val ifData = models.CrnMatchData("mycrn", "myname", models.Address("foo", "bar", None, None, "code"))
+
+        crnMatchingService.cleanMatch(knownFactsData, ifData) shouldBe true
+      }
+
+      "match where punctuation differs" in new Fixture {
+        val knownFactsData = models.CrnMatchData("myc!rn ", " my'name", models.Address("foo .", "b ar", None, None, "code"))
+        val ifData = models.CrnMatchData("mycrn", "myname", models.Address("foo", "bar", None, None, "code"))
+
+        crnMatchingService.cleanMatch(knownFactsData, ifData) shouldBe true
+      }
+
+      "match where PO box differs" in new Fixture {
+        val knownFactsData = models.CrnMatchData("myc!rn ", " my'name", models.Address("P.O. Box787", "bar", None, None, "code"))
```

Comment:
see above

Reply:
@parkeradam - Can do but not really a show stopper as each method builds on the last and the fact it passed means its cleaning up the PO Box correctly.